openmc/__init__.py

```diff
@@ -30,8 +30,10 @@
 from openmc.search import *
 from openmc.polynomial import *
 from . import examples
+from openmc.mpi import DummyCommunicator, MPI, comm
```

Comment:
I'm fine with `comm` moving to its own file, but I don't want it to be imported by default since that implies an import of `mpi4py` by default. Let's do the same approach as with lib/deplete -- if `comm` is needed, have a deferred import of `openmc.mpi` in the body of a function/method.

Reply:
Hmm, the presence of DummyCommunicator provides that protection via Mock and DummyCommunicator. Will take a look and see what the delayed import looks like.

Reply:
Yeah the ``comm`` object is only present throughout ``openmc.deplete`` and in a few places in ``model.py``. 

For ``openmc.deplete``, its so distributed by necessity that local imports will hinder more than help.

For ``model.py`` the place where its mostly needed is setting the default communicator in ``Model.__init__``. That would be used even if MPI functionality isn't desired. This could be moved elsewhere but then the Model won't know about the MPI state until some later point (probably init?), but I worry there could be some future system state manipulations which should be done in an MPI aware context that won't be. Could I tell you what those are now, not exactly, perhaps export_to_xml writes are one example but I'm sure thats not the best.

So, since the ``openmc.mpi`` package includes import protections already, I think this is probably the right approach.

Reply:
Let me restate just to make sure the concern is clear -- importing `mpi4py` actually results in `MPI_Init` getting called under the hood. You really don't want this to happen unless you are definitely going to be calling MPI code. As one example of what can go wrong, if you import `mpi4py` and then call `subprocess.run(['mpiexec', ...])`, it won't work for some MPI implementations. This would be a really annoying side effect of `import openmc` which is why I only want to import `mpi4py` if it is 100% needed.

Reply:
Ohhhhh thank you. That makes more sense now. I have corrected this behavior.